compound_disease/compound_treats_disease/literature_models/coco_score/coco_score.py

```diff
@@ -0,0 +1,376 @@
+# # CoCoScore Implementation
+
+# This notebook consists of implementing the [CoCoScore](https://www.biorxiv.org/content/10.1101/444398v1) literature model for comparison.
+
+# In[1]:
+
+
+get_ipython().run_line_magic('load_ext', 'autoreload')
+get_ipython().run_line_magic('autoreload', '2')
+get_ipython().run_line_magic('matplotlib', 'inline')
+
+from collections import defaultdict
+import os
+import pickle
+import re
+import sys
+
+sys.path.append(os.path.abspath('../../../../modules'))
+
+import operator
+from itertools import chain
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+from scipy import sparse
+from sklearn.linear_model import LogisticRegressionCV
+from sklearn.metrics import roc_curve, auc, f1_score, precision_recall_curve, accuracy_score, confusion_matrix
+from tqdm import tqdm_notebook
+
+
+# In[2]:
+
+
+#Set up the environment
+username = "danich1"
+password = "snorkel"
+dbname = "pubmeddb"
+
+#Path subject to change for different os
+database_str = "postgresql+psycopg2://{}:{}@/{}?host=/var/run/postgresql".format(username, password, dbname)
+os.environ['SNORKELDB'] = database_str
+
+from snorkel import SnorkelSession
+session = SnorkelSession()
+
+
+# In[3]:
+
+
+from snorkel.learning.pytorch.rnn.rnn_base import mark_sentence
+from snorkel.learning.pytorch.rnn.utils import candidate_to_tokens
+from snorkel.models import Candidate, candidate_subclass
+
+from metal.analysis import lf_summary
+from metal.label_model import LabelModel
+from metal.utils import plusminus_to_categorical
+
+from gensim.models import FastText
+from gensim.models import KeyedVectors
+
+from utils.notebook_utils.label_matrix_helper import label_candidates, get_auc_significant_stats
+from utils.notebook_utils.dataframe_helper import load_candidate_dataframes, generate_results_df
+from utils.notebook_utils.plot_helper import plot_curve, plot_label_matrix_heatmap
+
+
+# In[4]:
+
+
+CompoundDisease = candidate_subclass('CompoundDisease', ['Compound', 'Disease'])
+
+
+# In[5]:
+
+
+quick_load = True
+
+
+# In[6]:
+
+
+total_candidates_df = pd.read_table("../../dataset_statistics/results/all_ctd_map.tsv.xz")
+total_candidates_df.head(2)
+
+
+# In[7]:
+
+
+spreadsheet_names = {
+    'dev': '../../data/sentences/sentence_labels_dev.xlsx',
+    'test': '../../data/sentences/sentence_labels_test.xlsx'
+}
+
+
+# In[8]:
+
+
+candidate_dfs = {
+    key:load_candidate_dataframes(spreadsheet_names[key], "curated_ctd")
+    for key in spreadsheet_names
+}
+
+for key in candidate_dfs:
+    print("Size of {} set: {}".format(key, candidate_dfs[key].shape[0]))
+
+
+# In[9]:
+
+
+distant_supervision_marginals = pd.read_table("../../label_sampling_experiment/results/CtD/marginals/baseline_marginals.tsv.xz")
+distant_supervision_marginals.head(2)
+
+
+# In[10]:
+
+
+all_embedded_cd_df = pd.read_table("../../word_vector_experiment/results/all_embedded_cd_sentences.tsv.xz")
+all_embedded_cd_df.head(2)
+
+
+# In[11]:
+
+
+word_vectors = pd.read_csv(
```

Comment:
Why is the first row blank?

Reply:
Good catch that is a typo that I'll make sure to fix.